tests/test_strong_branching.py

```diff
@@ -0,0 +1,169 @@
+                downgain = max([down - lastlpobjval, 0])
+                upgain = max([up - lastlpobjval, 0])
+                scores[i] = self.scip.getBranchScoreMultiple(branch_cands[i], [downgain, upgain])
+                continue
+
+            # Strong branch!
+            down, up, downvalid, upvalid, downinf, upinf, downconflict, upconflict, lperror = self.scip.strongBranchVar(
+                branch_cands[i], 200, idempotent=self.idempotent)
+
+            # In the case of an LP error handle appropriately (for this example we just break the loop)
+            if lperror:
+                break
+
+            # In the case of both infeasible sub-problems cutoff the node
+            if not self.idempotent and downinf and upinf:
+                return {"result": SCIP_RESULT.CUTOFF}
+
+            # Calculate the gains for each up and down node that strong branching explored
+            if not downinf and downvalid:
+                down_bounds[i] = down
+                downgain = max([down - lpobjval, 0])
+            else:
+                downgain = 0
+            if not upinf and upvalid:
+                up_bounds[i] = up
+                upgain = max([down - lpobjval, 0])
+            else:
+                upgain = 0
+
+            # Update the pseudo-costs
+            if not self.idempotent:
+                lpsol = branch_cands[i].getLPSol()
+                if not downinf and downvalid:
+                    self.scip.updateVarPseudoCost(branch_cands[i], -self.scip.frac(lpsol), downgain, 1)
+                if not upinf and upvalid:
+                    self.scip.updateVarPseudoCost(branch_cands[i], 1 - self.scip.frac(lpsol), upgain, 1)
+
+            scores[i] = self.scip.getBranchScoreMultiple(branch_cands[i], [downgain, upgain])
+            if scores[i] > scores[best_cand_idx]:
+                best_cand_idx = i
+
+        # End strong branching
+        self.scip.endStrongBranching()
+
+        # In the case of an LP error
+        if lperror:
+            return {"result": SCIP_RESULT.DIDNOTRUN}
+
+        # Branch on the variable with the largest score
+        down_child, eq_child, up_child = self.model.branchVarVal(
+            branch_cands[best_cand_idx], branch_cands[best_cand_idx].getLPSol())
+
+        # Update the bounds of the down node and up node
+        if self.scip.allColsInLP() and not self.idempotent:
+            if down_child is not None and down_bounds[best_cand_idx] is not None:
+                self.scip.updateNodeLowerbound(down_child, down_bounds[best_cand_idx])
+            if up_child is not None and up_bounds[best_cand_idx] is not None:
+                self.scip.updateNodeLowerbound(up_child, up_bounds[best_cand_idx])
+
+        return {"result": SCIP_RESULT.BRANCHED}
+
+
+def create_model():
+    scip = Model()
+    scip.setHeuristics(SCIP_PARAMSETTING.OFF)
+    scip.setSeparating(SCIP_PARAMSETTING.OFF)
+    # scip.setIntParam("presolving/maxrounds", 0)
```

Comment:
This commented code might have been forgotten

Reply:
Yup. Have removed.